src/SFML/Graphics/Image.cpp

```diff
@@ -55,14 +55,15 @@ namespace
 // stb_image callbacks that operate on a sf::InputStream
 int read(void* user, char* data, int size)
 {
-    auto& stream = *static_cast<sf::InputStream*>(user);
-    return static_cast<int>(stream.read(data, size));
+    auto&      stream = *static_cast<sf::InputStream*>(user);
+    const auto count  = stream.read(data, static_cast<std::size_t>(size));
+    return count ? static_cast<int>(*count) : -1;
```

Comment:
```suggestion
    return static_cast<sf::InputStream*>(user)->read(data, static_cast<std::size_t>(size)).value_or(-1);
```

Or keep the `stream` variable here if the line seems too complicated.

Reply:
std::optional<std::size_t>::value_or returns a std::size_t so .value_or(-1) will return a std::numeric_limits<std::size_t>::max() which is not what I want to do here.

Reply:
Same readability concern